include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -600,6 +600,79 @@ using __repacked_tuple_t = typename __repacked_tuple<T>::type;
 template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
+//-----------------------------------------------------------------------
+// future and helper classes for async pattern/algorithm
+//-----------------------------------------------------------------------
+
+// empty base class for type erasure
+struct __tmp_base
+{
+};
+
+// derived class to keep temporaries (e.g. buffer) alive
+template <typename... Ts>
+struct __TempObjs : public __tmp_base
```

Comment:
All the types in the patch a named in snake_case. Only this one is named with PascalCase.

More important: the name doesn't reflect the purpose of the type. I would name it something like `lifetime_extender` or `lifetime_keeper` or `objects_keeper`. Please choose whatever you like or propose yours.

Please rename `tmp_base` correspondently.

Reply:
Makes sense. Done.